stdlib/src/collections/list.mojo

```diff
@@ -239,6 +239,73 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             earlier_idx -= 1
             later_idx -= 1
 
+    @always_inline
+    fn __mul(inout self, x: Int):
+        """Appends the original elements of this list x-1 times.
+
+        ```mojo
+        var a = List[Int](1, 2)
+        a.__mul(2) # a = [1, 2, 1, 2]
+        ```
+
+        Args:
+            x: The multiplier number.
+        """
+        if x == 0:
+            self.clear()
+        var orig = List(self)
```

Comment:
Should probably create a `List` with precomputed capacity to prevent allocation.